### PR TITLE
feat: Create dataset header component

### PR DESCRIPTION
**Pull request by lyndsiWilliams**
### SUMMARY

This is the blank state implementation of the header component for the create dataset page. This PR also adds an option tooltip to the button in the `PageHeaderWithActions` component.

#### BEFORE:
layoutWithHeaderBefore

#### AFTER:
layoutWithHeaderAfter
headerAfter

### TESTING INSTRUCTIONS

- Go to `http://localhost:9000/dataset/add/?testing`
- Observe the header component
- Click the dataset name textbox and change the text by either pressing `enter` or clicking out of the box
  - Observe the text change
- Mousehover over the disabled save and menu buttons on the left
  - Observe the tooltip

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx [hunk 1]
```diff
@@ -152,6 +158,8 @@ export const PageHeaderWithActions = ({
               css={menuTriggerStyles}
               buttonStyle="tertiary"
               aria-label={t('Menu actions trigger')}
+              tooltip={tooltipProps?.text ?? undefined}
+              placement={tooltipProps?.placement ?? undefined}
```
Comment (geido) [inline superset-frontend/src/components/PageHeaderWithActions/index.tsx:162]:
```suggestion
              tooltip={tooltipProps?.text}
              placement={tooltipProps?.placement}
```

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import React, { ReactNode, ReactElement } from 'react';
 import { css, SupersetTheme, t, useTheme } from '@superset-ui/core';
 import { AntdDropdown, AntdDropdownProps } from 'src/components';
+import { TooltipPlacement } from 'antd/es/tooltip';
```
Comment (geido) [inline superset-frontend/src/components/PageHeaderWithActions/index.tsx:22]: Would you mind exporting this from the Tooltip component in our components library and then importing it here? I think we should centralize all the components related things and avoid accessing antd directly

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/AddDataset.test.tsx [hunk 3]
```diff
@@ -17,17 +17,28 @@
...
  */
 import React from 'react';
-import { render, screen } from 'spec/helpers/testing-library';
+import { render, screen, act } from 'spec/helpers/testing-library';
 import AddDataset from 'src/views/CRUD/data/dataset/AddDataset';
 
 describe('AddDataset', () => {
-  it('renders a blank state AddDataset', () => {
-    render(<AddDataset />);
+  const renderAndWait = async () => {
```
Comment (geido) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/AddDataset.test.tsx:24]: If the purpose here to wait until the component has finished loading, you can use something like this

```
const waitForRender = (overrides: Record<string, any> = {}) =>
  waitFor(() => render(<AddDataset />);

await waitForRender();
```

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx [hunk 4]
```diff
@@ -17,7 +17,85 @@
...
+import { t, styled, css, SupersetTheme } from '@superset-ui/core';
+import { PageHeaderWithActions } from 'src/components/PageHeaderWithActions';
+import Button from 'src/components/Button';
+import Icons from 'src/components/Icons';
+import { Menu } from 'src/components/Menu';
+import { TooltipPlacement } from 'antd/es/tooltip';
+
+const editableTitleProps = {
+  title: '',
+  placeholder: 'Add the name of the dataset',
```
Comment (geido) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx:29]: I think all the text in this file needs to be localized

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx [hunk 5]
```diff
@@ -17,7 +17,85 @@
...
+  canEdit: true,
+  label: 'dataset name',
+};
+
+const tooltipProps: { text: string; placement: TooltipPlacement } = {
+  text: 'Select a database table and create dataset',
+  placement: 'bottomRight',
+};
+
+const Styles = styled.div`
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx:40]: should this go into styles?

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx [hunk 6]
```diff
@@ -17,7 +17,85 @@
...
+
+    &:disabled {
+      svg {
+        color: ${({ theme }) => theme.colors.grayscale.light1};
+      }
+    }
+  }
+`;
+
+const disabledSaveBtnStyles = (theme: SupersetTheme) => css`
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx:54]: same with this.

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx [hunk 7]
```diff
@@ -17,7 +17,85 @@
...
+const renderOverlay = () => (
+  <Menu>
+    <Menu.Item>{t('Settings')}</Menu.Item>
+    <Menu.Item>{t('Delete')}</Menu.Item>
+  </Menu>
+);
 
 export default function Header() {
-  return <div>Header</div>;
+  return (
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx:84]: hey, quick question in regards to the save button. What happens, currently, when I press save? I believe that we have a state action that is supposed to put the name into state. Does it currently do that?

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx (hunk 1 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/components/PageHeaderWithActions/index.tsx]: Fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/components/PageHeaderWithActions/index.tsx (hunk 2 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/components/PageHeaderWithActions/index.tsx]: Good thinking! Fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/AddDataset.test.tsx (hunk 3 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/AddDataset.test.tsx]: Oh yeah that's a lot cleaner! Fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx (hunk 4 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx]: Good call, fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx (hunk 5 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx]: Oh yeah good call, fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx (hunk 6 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx]: Also fixed in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx (hunk 7 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/index.tsx:80]: Good call, I have the data properly flowing now in [`this commit`](https://github.com/apache/superset/pull/21189/commits/890af00638acaa42d9f7d35bd86f3ad573739122).

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/Header.test.tsx [hunk 8]
```diff
@@ -17,13 +17,34 @@
...
+    });
+    const menuButton = screen.getByRole('button', {
+      name: /menu actions trigger/i,
+    });
+
+    expect(datasetNameTextbox).toBeVisible();
+    expect(saveButton).toBeVisible();
+    expect(saveButton).toBeDisabled();
+    expect(menuButton).toBeVisible();
+    expect(menuButton).toBeDisabled();
```
Comment (AAfghahi) [inline superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/Header.test.tsx:48]: should there be a test for when someone passes in a name?

**Review by AAfghahi — APPROVED** [review]

File: superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/Header.test.tsx (hunk 8 above)
Comment (lyndsiWilliams) [reply superset-frontend/src/views/CRUD/data/dataset/AddDataset/Header/Header.test.tsx:49]: Can do! Added in [`this commit`](https://github.com/apache/superset/pull/21189/commits/0ba4ab1da0052e3431b2f0ddb649d5fcd4e28503).

File: superset-frontend/src/views/CRUD/data/dataset/DatasetLayout/DatasetLayout.test.tsx [hunk 9]
```diff
@@ -33,10 +33,21 @@ describe('DatasetLayout', () => {
...
   });
 
-  it('renders a Header when passed in', () => {
-    render(<DatasetLayout header={Header()} />);
+  const mockSetDataset = jest.fn();
 
-    expect(screen.getByText(/header/i)).toBeVisible();
+  const waitForRender = () =>
+    waitFor(() =>
+      render(<Header setDataset={mockSetDataset} datasetName="" />),
```
Comment (hughhhh) [inline superset-frontend/src/views/CRUD/data/dataset/DatasetLayout/DatasetLayout.test.tsx:40]: nit: add a test to make sure if `datasetName` is set that value of the textbox renders

**Comment by jinghua-qa** [comment]
LGTM

**Review by hughhhh — APPROVED** [review]

**Comment by C-monC** [comment]
Any possibility this induces https://github.com/apache/superset/issues/21476?

**Pull request merged**